eks_infrastructure/patch_hosts.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#/ Usage: ./patch_hosts.sh
+
+set -ex
+source ./helper.sh
+
+# Parse parameters from build_param.json config file
+EKS_CLUSTERS=($(jq -r '.eks_clusters[]' build_param.json))
+CONTEXTS=($(jq -r '.contexts[]' build_param.json))
+EKS_VERSION=$(jq -r '.eks_version' build_param.json)
+CLUSTER_AUTOSCALAR_IMAGE_VERSION=$(jq -r '.cluster_autoscalar_image_version' build_param.json)
+
+# Create a cluster if the context is predeploy and operation is other than create
+if [ "${CONTEXTS}" = "BETA" ] && [ "${OPERATION}" != "create" ]; then
```

Comment:
Either we keep this condition, in which case a staging cluster needs to be added to buildspec. If we do not want to test the upgrade operation on staging cluster, this condition and the delete condition at the end of the file can be removed.

I am inclined on not having staging cluster is because when a new cluster is created the AMI is used as the latest, hence performing upgrade nodegroup has no effect.

Reply:
I agree with no having this stage instead. I meant to delete that in the previous commit that I deleted the input logic but forgot to delete the staging logic.